File: 투 포인터/2470/2470.py
n = int(input())
m = 2000000000
arr = list(map(int, input().split()))

arr.sort()
l, r = 0, n - 1
a, b = 0, 0
while l < r:
   if m > abs(arr[l] + arr[r]):
      a, b = arr[l], arr[r]
      m = abs(arr[l] + arr[r])
   if arr[l] + arr[r] < 0:
      l += 1
   elif arr[l] + arr[r] > 0:
      r -= 1
   else:
      break
print(a, b)

File: 투 포인터/2470/2470.cpp
#include <algorithm>
#include <iostream>
#include <vector>
using namespace std;

int main()
{
   int n, l, r, a, b;
   int m = 2000000000;
   cin >> n;
   vector<int> v(n);
   for (int i = 0; i < n; ++i)
      cin >> v[i];

   sort(v.begin(), v.end());
   l = 0, r = n - 1;
   while (l < r)
   {
      if (m > abs(v[l] + v[r]))
      {
         a = v[l], b = v[r];
         m = abs(v[l] + v[r]);
      }
      if (v[l] + v[r] < 0)
         ++l;
      else if (v[l] + v[r] > 0)
         --r;
      else
         break;
   }
   cout << a << " " << b;

   return 0;
}